### PR TITLE
[alamin step-6] Model creations

### PR DESCRIPTION
**Pull request by rr-mdalauddin-alamin**
### Step 6: Let's create a task model

Create CRUD to manage the tasks. First of all, let's make it with a simple structure where only the name and details can be registered.

- Create task model for CRUD with `rails generate`
- Use migration and create tables
  - It is important to ensure that the migration can be returned to the previous state! Let's get into the habit of checking by using `redo`
- Make sure you can connect to the database via the model with the command `rails c`
  - Try creating a record with ActiveRecord
- Create a PR on GitHub for review

### ERD for reference
ERD v1

### CODE REVIEW

**Comment by Katsumi-N** [comment]
Could you put ER diagram on this PR description?

**Comment by rr-mdalauddin-alamin** [comment]
Sure! Added

File: myapp/app/models/task.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Task < ApplicationRecord
+
+  STATUSES = ["Not Started", "In Progress", "Completed"].freeze
+
+  belongs_to :user
+  belongs_to :user, :column :assigned_user_id
+
+  has_many :labels, through: :tasks_labels
```
Comment (Katsumi-N) [inline myapp/app/models/task.rb:10]: `has_many :tasks_labels` is also necessary, maybe

File: myapp/app/models/task.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class Task < ApplicationRecord
+
+  STATUSES = ["Not Started", "In Progress", "Completed"].freeze
```
Comment (Katsumi-N) [inline myapp/app/models/task.rb]: [FYI] You can use `enum`
ex
```
enum status: { not_started: 0, in_progress: 1, completed: 2 }
```
https://blog.saeloun.com/2022/01/05/how-to-use-enums-in-rails/
https://edgeapi.rubyonrails.org/classes/ActiveRecord/Enum.html

File: myapp/app/models/task.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+  belongs_to :user
+  belongs_to :user, :column :assigned_user_id
+
+  has_many :labels, through: :tasks_labels
+
+  validates :title, presence: true
+  validates :status, presence: true, inclusion: { in: STATUSES }
+  validates :priority, presence: true,
```
Comment (Katsumi-N) [inline myapp/app/models/task.rb:12]: Please remove comma of last line as it remains
(maybe you got error when migrating?)

File: myapp/app/models/task.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+class Task < ApplicationRecord
+
+  STATUSES = ["Not Started", "In Progress", "Completed"].freeze
+
+  belongs_to :user
+  belongs_to :user, :column :assigned_user_id
```
Comment (Katsumi-N) [inline myapp/app/models/task.rb]: I don't know `column` param(and can't find document)
I think you can modify like this
```
belongs_to :assigned_user_id, class_name: 'User', foreign_key: 'user_id'
```

**Review by Katsumi-N — COMMENTED** [review]
Could you please attach a brief operational check?
For example, create and retrieve users/tasks in rails console

File: myapp/app/models/task.rb (hunk 4 above)
Comment (rr-mdalauddin-alamin) [reply myapp/app/models/task.rb]: Could you give me the url that you found it.
`belongs_to :assigned_user_id, class_name: 'User', foreign_key: 'user_id'`
Its not actually working in my console.

Comment (Katsumi-N) [reply myapp/app/models/task.rb]: I found this url
https://guides.rubyonrails.org/association_basics.html#the-belongs-to-association:~:text=For%20example%2C%20consider%20the%20following%20model%20declarations%3A

Comment (rr-mdalauddin-alamin) [reply myapp/app/models/task.rb]: got a workaround, thanks for the clue ;)

File: myapp/app/models/task.rb (hunk 1 above)
Comment (rr-mdalauddin-alamin) [reply myapp/app/models/task.rb:10]: added!

File: myapp/app/models/task.rb (hunk 2 above)
Comment (rr-mdalauddin-alamin) [reply myapp/app/models/task.rb]: modified

**Review by Katsumi-N — APPROVED** [review]
LGTM!
When you modify after review, you may divide each commit so that reviewer can easily review .
Could you do it from next step? 😃

**Comment by rr-mdalauddin-alamin** [comment]
@Katsumi-N 
Sure! No problem!
Thank you!

**Pull request merged**